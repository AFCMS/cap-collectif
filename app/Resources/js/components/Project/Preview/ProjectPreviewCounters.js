// @flow
import React from 'react';
import ProjectPreviewCounter from './ProjectPreviewCounter';

const ProjectPreviewCounters = React.createClass({
  propTypes: {
    project: React.PropTypes.object.isRequired,
  },

  getNbCounters() {
    const { project } = this.props;
    const votesCount = project.votesCount;
    let nb = 2;
    nb += votesCount ? 1 : 0;
    return nb;
  },

  render() {
    const { project } = this.props;
    const nbCounters = this.getNbCounters();
    const counterWidth = `${100 / nbCounters}%`;
    return (
      <div className="thumbnail__numbers">
        <ProjectPreviewCounter
          value={project.contributionsCount}
          label="project.preview.counters.contributions"
          style={{ width: counterWidth }}
          showZero
        />
        <ProjectPreviewCounter
          value={project.votesCount}
          label="project.preview.counters.votes"
          style={{ width: counterWidth }}
        />
        <ProjectPreviewCounter
          value={project.participantsCount}
          label="project.preview.counters.contributors"
          style={{ width: counterWidth }}
          showZero
        />
      </div>
    );
  },
});

export default ProjectPreviewCounters;
