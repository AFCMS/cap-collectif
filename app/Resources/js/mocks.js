// @flow
/* eslint-env jest */
import { type FormProps } from 'redux-form';
import { type IntlShape } from 'react-intl';
import type { RelayPaginationProp, RelayRefetchProp } from 'react-relay';

export const $refType: any = null;
export const $fragmentRefs: any = null;

export const intlMock: IntlShape = {
  locale: 'fr-FR',
  formats: {},
  messages: {},
  now: () => 0,
  formatHTMLMessage: message => String(message),
  formatPlural: message => String(message),
  formatNumber: message => String(message),
  formatRelative: message => String(message),
  formatTime: message => String(message),
  formatDate: message => String(message),
  formatMessage: message => String(message.id),
};

export const formMock: FormProps = {
  anyTouched: false,
  array: {
    insert: jest.fn(),
    move: jest.fn(),
    pop: jest.fn(),
    push: jest.fn(),
    remove: jest.fn(),
    removeAll: jest.fn(),
    shift: jest.fn(),
    splice: jest.fn(),
    swap: jest.fn(),
    unshift: jest.fn(),
  },
  asyncValidate: jest.fn(),
  asyncValidating: false,
  autofill: jest.fn(),
  blur: jest.fn(),
  change: jest.fn(),
  clearAsyncError: jest.fn(),
  clearSubmit: jest.fn(),
  destroy: jest.fn(),
  dirty: false,
  dispatch: jest.fn(),
  error: null,
  form: 'formName',
  handleSubmit: jest.fn(),
  initialize: jest.fn(),
  initialized: true,
  initialValues: {},
  invalid: false,
  pristine: true,
  reset: jest.fn(),
  submitting: false,
  submitFailed: false,
  submitSucceeded: false,
  touch: jest.fn(),
  untouch: jest.fn(),
  valid: true,
  warning: null,
};

export const relayPaginationMock: RelayPaginationProp = {
  environment: {
    applyMutation: jest.fn(),
    sendMutation: jest.fn(),
    lookup: jest.fn(),
    sendQuery: jest.fn(),
    subscribe: jest.fn(),
    streamQuery: jest.fn(),
    retain: jest.fn(),
    unstable_internal: {},
  },
  hasMore: () => false,
  isLoading: () => false,
  loadMore: jest.fn(),
  refetchConnection: jest.fn(),
};

export const relayRefetchMock: RelayRefetchProp = {
  environment: {
    applyMutation: jest.fn(),
    sendMutation: jest.fn(),
    lookup: jest.fn(),
    sendQuery: jest.fn(),
    subscribe: jest.fn(),
    streamQuery: jest.fn(),
    retain: jest.fn(),
    unstable_internal: {},
  },
  refetch: jest.fn(),
};
