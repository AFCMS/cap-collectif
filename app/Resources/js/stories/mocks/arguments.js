// @flow
import { author } from './users';

export const argument = {
  user: author,
  trashedStatus: '',
  createdAt: '1 mars 2016 à 00:00',
  publishedAt: '1 mars 2016 à 00:00',
  body:
    'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: { totalCount: 50 },
};

export const argumentsData = [
  { ...argument, user: { ...argument.user, vip: true } },
  { ...argument },
  { ...argument, trashedStatus: 'INVISIBLE' },
  { ...argument },
];
