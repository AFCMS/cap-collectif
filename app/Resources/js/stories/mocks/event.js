// @flow
import { author } from './users';

export const event = {
  title: 'PHPTourDuTurfu',
  url: '#',
  image: {
    url: 'https://picsum.photos/300/400',
    alt: 'event picture',
  },
  date: {
    start: '2030-03-10 00:00:00',
    end: '2032-06-15 00:00:00',
  },
  location: {
    place: 'Tour Eiffel',
    zip: '75007',
    city: 'Paris',
  },
  category: 'Immobilier',

  user: author,
};
