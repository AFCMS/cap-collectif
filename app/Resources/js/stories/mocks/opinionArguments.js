// @flow
import { author } from './users';

export const opinionArgument = {
  user: author,
  trashedStatus: '',
  createdAt: '1 mars 2016 à 00:00',
  publishedAt: '1 mars 2016 à 00:00',
  body:
    'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: { totalCount: 50 },
};

export const opinionArguments = [
  { ...opinionArgument, user: { ...opinionArgument.user, vip: true } },
  { ...opinionArgument },
  { ...opinionArgument, trashedStatus: 'INVISIBLE' },
  { ...opinionArgument },
];
