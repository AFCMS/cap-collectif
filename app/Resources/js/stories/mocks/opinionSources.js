// @flow
import { author } from './users';

export const opinionSource = {
  url: 'https://ui.cap-collectif.com/',
  user: author,
  createdAt: ' • 1 mars 2016',
  publishedAt: ' • 1 mars 2016',
  title: 'Source title',
  category: 'Politique',
  body:
    'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: { totalCount: 2 },
};

export const opinionSources = [
  { ...opinionSource },
  { ...opinionSource },
  { ...opinionSource, user: { ...opinionSource.user, vip: true } },
  { ...opinionSource },
  { ...opinionSource, category: null },
];
