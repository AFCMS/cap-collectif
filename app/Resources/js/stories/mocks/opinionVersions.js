// @flow
import { author } from './users';

export const opinionVersion = {
  url: 'https://ui.cap-collectif.com/',
  title: 'Opinion version',
  user: author,
  createdAt: ' • 3 janvier 2015',
  pinned: true,
  publishedAt: ' • 3 janvier 2015',
  votes: { totalCount: 50 },
  arguments: { totalCount: 4 },
  sources: { totalCount: 0 },
  votesMitige: { totalCount: 1 },
  votesNok: { totalCount: 1 },
  votesOk: { totalCount: 2 },
};

export const opinionVersions = [
  { ...opinionVersion },
  { ...opinionVersion, user: { ...opinionVersion.user, vip: true } },
  { ...opinionVersion, votes: { totalCount: 0 } },
];
