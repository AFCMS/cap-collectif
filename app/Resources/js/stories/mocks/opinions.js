// @flow
import { author } from './users';

export const opinion = {
  url: 'https://ui.cap-collectif.com/',
  title: 'Opinion',
  user: author,
  createdAt: ' • 1 mars 2018',
  pinned: true,
  publishedAt: ' • 1 mars 2018',
  votes: { totalCount: 4 },
  versions: { totalCount: 3 },
  arguments: { totalCount: 4 },
  sources: { totalCount: 0 },
  votesMitige: { totalCount: 1 },
  votesNok: { totalCount: 1 },
  votesOk: { totalCount: 2 },
};

export const opinions = [
  { ...opinion },
  { ...opinion, user: { ...opinion.user, vip: true } },
  { ...opinion, votes: { totalCount: 0 } },
];
