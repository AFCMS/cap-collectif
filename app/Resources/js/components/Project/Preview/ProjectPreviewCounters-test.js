// @flow
/* eslint-env jest */
import * as React from 'react';
import { shallow } from 'enzyme';

import { $refType, $fragmentRefs } from '../../../mocks';
import { ProjectPreviewCounters } from './ProjectPreviewCounters';

const props = {
  project: {
    $fragmentRefs,
    $refType,
    id: 'UHJvamVjdDpwcm9qZWN0MQ==',
    districts: [],
    contributors: { totalCount: 48, anonymousCount: 2 },
    votes: { totalCount: 48 },
    contributionsCount: 71,
    isVotesCounterDisplayable: true,
    isContributionsCounterDisplayable: true,
    isParticipantsCounterDisplayable: true,
  },
};

const propsWithDistricts = {
  project: {
    $fragmentRefs,
    $refType,
    id: 'project2',
    districts: [{ name: 'rive gauche' }, { name: 'rive droite' }],
    contributors: { totalCount: 54, anonymousCount: 0 },
    votes: { totalCount: 54 },
    contributionsCount: 89,
    isVotesCounterDisplayable: true,
    isContributionsCounterDisplayable: true,
    isParticipantsCounterDisplayable: true,
  },
};

describe('<ProjectPreviewProgressBar />', () => {
  it('should render correctly counters without districts', () => {
    const wrapper = shallow(<ProjectPreviewCounters {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly counters with districts', () => {
    const wrapper = shallow(<ProjectPreviewCounters {...propsWithDistricts} />);
    expect(wrapper).toMatchSnapshot();
  });
});
