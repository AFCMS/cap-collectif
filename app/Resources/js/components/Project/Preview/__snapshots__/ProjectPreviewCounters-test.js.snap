// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectPreviewProgressBar /> should render correctly counters with districts 1`] = `
<TagsList>
  <ProjectPreviewCounter
    icon="cap-baloon-1"
    label="project.preview.counters.contributions"
    showZero={true}
    value={89}
  />
  <ProjectPreviewCounter
    icon="cap-hand-like-2-1"
    label="project.preview.counters.votes"
    showZero={false}
    value={54}
  />
  <ProjectPreviewCounter
    icon="cap-user-2-1"
    label="project.preview.counters.contributors"
    showZero={true}
    value={54}
  />
  <Tag
    icon="cap cap-marker-1-1"
  >
    <InlineList
      className="d-i"
      margin={true}
      separator=","
    >
      <li
        key="0"
      >
        rive gauche
      </li>
      <li
        key="1"
      >
        rive droite
      </li>
    </InlineList>
  </Tag>
  <ForwardRef(Relay(ProjectRestrictedAccessFragment))
    icon="cap-lock-2-1"
    project={
      Object {
        "$fragmentRefs": null,
        "$refType": null,
        "contributionsCount": 89,
        "contributors": Object {
          "anonymousCount": 0,
          "totalCount": 54,
        },
        "districts": Array [
          Object {
            "name": "rive gauche",
          },
          Object {
            "name": "rive droite",
          },
        ],
        "id": "project2",
        "isContributionsCounterDisplayable": true,
        "isParticipantsCounterDisplayable": true,
        "isVotesCounterDisplayable": true,
        "votes": Object {
          "totalCount": 54,
        },
      }
    }
  />
</TagsList>
`;

exports[`<ProjectPreviewProgressBar /> should render correctly counters without districts 1`] = `
<TagsList>
  <ProjectPreviewCounter
    icon="cap-baloon-1"
    label="project.preview.counters.contributions"
    showZero={true}
    value={71}
  />
  <ProjectPreviewCounter
    icon="cap-hand-like-2-1"
    label="project.preview.counters.votes"
    showZero={false}
    value={48}
  />
  <ProjectPreviewCounter
    icon="cap-user-2-1"
    label="project.preview.counters.contributors"
    showZero={true}
    value={50}
  />
  <ForwardRef(Relay(ProjectRestrictedAccessFragment))
    icon="cap-lock-2-1"
    project={
      Object {
        "$fragmentRefs": null,
        "$refType": null,
        "contributionsCount": 71,
        "contributors": Object {
          "anonymousCount": 2,
          "totalCount": 48,
        },
        "districts": Array [],
        "id": "UHJvamVjdDpwcm9qZWN0MQ==",
        "isContributionsCounterDisplayable": true,
        "isParticipantsCounterDisplayable": true,
        "isVotesCounterDisplayable": true,
        "votes": Object {
          "totalCount": 48,
        },
      }
    }
  />
</TagsList>
`;
